src/Widgets/SaveDialog.vala

```diff
@@ -131,8 +131,45 @@ public class Screenshot.SaveDialog : Granite.Dialog {
         var location_label = new Gtk.Label (_("Folder:"));
         location_label.halign = Gtk.Align.END;
 
-        var location = new Gtk.FileChooserButton (_("Select Screenshots Folder…"), Gtk.FileChooserAction.SELECT_FOLDER);
-        location.set_current_folder (folder_dir);
+        var folder_name = new Gtk.Label ("") {
+            halign = Gtk.Align.START
+        };
+
+        var folder_icon = new Gtk.Image () {
+            pixel_size = 16
+        };
+
+        if (folder_dir == Environment.get_home_dir ()) {
+            folder_name.label = "Home";
+            folder_icon.gicon = new ThemedIcon ("user-home");
+        } else if (folder_dir == "/") {
+            folder_name.label = "File System";
+            folder_icon.gicon = new ThemedIcon ("drive-harddisk");
+        } else {
+            folder_name.label = folder_dir.reverse ().split ("/")[0].reverse ();
+            folder_icon.gicon = new ThemedIcon ("folder");
+        }
+
+        var arrow = new Gtk.Image () {
+            gicon = new ThemedIcon ("pan-down-symbolic"),
+            halign = Gtk.Align.END
+        };
+
+        var location_button_indicator = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 0);
+        location_button_indicator.pack_start (folder_icon, false, false);
+        location_button_indicator.pack_start (folder_name, true, true);
+        location_button_indicator.pack_start (arrow, false, false);
+
+        var location_button = new Gtk.Button ();
+        location_button.add (location_button_indicator);
+
+        var location_dialog = new Gtk.FileChooserNative (_("Select Screenshots Folder…"), this,
+            Gtk.FileChooserAction.SELECT_FOLDER, "Open", "Cancel");
+        location_dialog.set_current_folder (folder_dir);
+
+        var size_group = new Gtk.SizeGroup (Gtk.SizeGroupMode.VERTICAL);
+        size_group.add_widget (format_cmb);
+        size_group.add_widget (location_button);
 
```

Comment:
You can avoid using a size group by taking the `preview_box` out of the grid and then applying `row_homogeneous = true` to the grid (and not making the top margin zero).

Reply:
Great Idea. Thanks